tutorial/numpy-functions.md

```diff
@@ -0,0 +1,350 @@
+        }
+        imag_part = array_create(1, (int64_t[]){4}, nd_double);
+        for (i_0001 = 0; i_0001 < 4; i_0001 += 1)
+        {
+            imag_part.nd_double[get_index(imag_part, i_0001)] = cimag(arr1.nd_cdouble[get_index(arr1, i_0001)]);
+        }
+        printf("%s ", "real part for arr1: ");
+        printf("%s", "[");
+        for (i = 0; i < 4 - 1; i += 1)
+        {
+            printf("%.12lf ", real_part.nd_double[get_index(real_part, i)]);
+        }
+        printf("%.12lf]", real_part.nd_double[get_index(real_part, 4 - 1)]);
+        printf("%s ", "\nimag part for arr1: ");
+        printf("%s", "[");
+        for (i_0002 = 0; i_0002 < 4 - 1; i_0002 += 1)
+        {
+            printf("%.12lf ", imag_part.nd_double[get_index(imag_part, i_0002)]);
+        }
+        printf("%.12lf]\n", imag_part.nd_double[get_index(imag_part, 4 - 1)]);
+        free_array(arr1);
+        free_array(real_part);
+        free_array(imag_part);
+        return 0;
+    }
+    ```
+
+## [Prod](https://numpy.org/doc/stable/reference/generated/numpy.prod.html)
+
+-   Supported parameters:
+
+    a: array_like,
+        Input data.
+
+-   Supported languages: fortran
+
+-   python code:
+
+    ```python
+    from numpy import array, prod
+
+    arr = array([1,2,3,4])
+    prd = prod(arr)
+    print("prd: ", prd)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_prod
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:)
+    integer(C_INT64_T) :: prd
+
+    allocate(arr(0:4_C_INT64_T - 1_C_INT64_T))
+    arr = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    prd = product(arr)
+    print *, 'prd: ' // ' ' , prd
+
+    end program prog_test_prod
+    ```
+
+## [mod](https://numpy.org/doc/stable/reference/generated/numpy.mod.html)
+
+-   Supported parameters:
+
+    x1: array_like
+        Dividend array.
+
+    x2: array_like,
+        Divisor array. If x1.shape != x2.shape, they must be
+        broadcastable to a common shape (which becomes the shape of the output).
+
+-   Supported language: fortran.
+
+-   python code:
+
+    ```python
+    from numpy import array, mod
+
+    arr = array([1,2,3,4])
+    res = mod(arr, arr)
+    print("res: ", res)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_mod
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:)
+    integer(C_INT64_T), allocatable :: res(:)
+
+    allocate(arr(0:4_C_INT64_T - 1_C_INT64_T))
+    arr = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    allocate(res(0:4_C_INT64_T - 1_C_INT64_T))
+    res = modulo(arr,arr)
+    print *, 'res: ' // ' ' , res
+
+    end program prog_test_mod
+    ```
+
+## [matmul](https://numpy.org/doc/stable/reference/generated/numpy.matmul.html)
+
+-   Supported parameters:
+
+    x1, x2: array_like,
+        Input arrays, scalars not allowed.
```

Comment:
I think you need to specify that these arrays must be 2d

Reply:
i think we should just mention that they should be vectors or matrixes.

Reply:
Ah, I didn't realise it worked for vectors :)

It definitely doesn't work for n-d arrays with n>2 though, so that needs mentioning

Reply:
Are you sure it works for vectors? All the tests seem to be marked xfail

Reply:
yeah, we have opened an issue about that, but for python it works.

Reply:
Nobody uses pyccel to translate python to python. That is supported internally to ensure that code which creates pyccel nodes (e.g. psydac) can print to python. So for 99.9% of users, it doesn't work. So I don't think it's a good idea to claim that it does here.

Alternatively, you can write:
`Supported langauges: fortran (2d only)`

Reply:
i meant for python itself, not the generated code.